GAD 8 - Binomial Heap/tests/binomilia/BinomialTreeNodeTester.java

```diff
@@ -0,0 +1,213 @@
+        assertEquals(50, treeNode.min());
+        assertEquals(1, treeNode.rank());
+
+        BinomialTreeNode.merge(treeNode, BinomialTreeNode.merge(new BinomialTreeNode(60), new BinomialTreeNode(69)));
+        assertEquals(50, treeNode.min());
+        assertEquals(2, treeNode.rank());
+
+        BinomialTreeNode.merge(treeNode, BinomialTreeNode.merge(
+                BinomialTreeNode.merge(new BinomialTreeNode(700), new BinomialTreeNode(800)),
+                BinomialTreeNode.merge(new BinomialTreeNode(6969), new BinomialTreeNode(420))
+        ));
+        assertEquals(50, treeNode.min());
+        assertEquals(3, treeNode.rank());
+
+
+        BinomialTreeNode smallerTree = new BinomialTreeNode(-10);
+
+        assertEquals(-10, smallerTree.min());
+
+        BinomialTreeNode.merge(smallerTree, new BinomialTreeNode(-6));
+        assertEquals(-10, smallerTree.min());
+        assertEquals(1, smallerTree.rank());
+
+        BinomialTreeNode.merge(smallerTree, BinomialTreeNode.merge(new BinomialTreeNode(25), new BinomialTreeNode(10)));
+        assertEquals(-10, smallerTree.min());
+        assertEquals(2, smallerTree.rank());
+
+        BinomialTreeNode.merge(smallerTree, BinomialTreeNode.merge(
+                BinomialTreeNode.merge(new BinomialTreeNode(22), new BinomialTreeNode(0)),
+                BinomialTreeNode.merge(new BinomialTreeNode(12), new BinomialTreeNode(100000))
+        ));
+        assertEquals(-10, smallerTree.min());
+        assertEquals(3, smallerTree.rank());
+
+        BinomialTreeNode.merge(smallerTree, treeNode);
+        assertEquals(-10, smallerTree.min());
+        assertEquals(50, treeNode.min());
+        assertEquals(4, smallerTree.rank());
+        assertEquals(3, treeNode.rank());
+
+    }
+
+    /**
+     * Relatively complex tests to see if {@link BinomialTreeNode#merge(BinomialTreeNode, BinomialTreeNode)} returns
+     * the correct node as well as if the ranks and mins of the returned nodes are being changed or not. <br>
+     * This test uses all the methods of {@link BinomialTreeNode}
+     * @author Aamin
+     */
+    @Test
+    public void merge() {
+        BinomialTreeNode treeNode1 = new BinomialTreeNode(50);
+        BinomialTreeNode treeNode2 = new BinomialTreeNode(69);
+
+        testMinAndRank();
+
+        assertEquals(treeNode1, BinomialTreeNode.merge(treeNode1, treeNode2), "You returned the wrong node!");
+        assertEquals(1, treeNode1.rank());
+        assertEquals(50, treeNode1.min());
+
+        BinomialTreeNode treeNode3 = new BinomialTreeNode(12);
+        BinomialTreeNode treeNode4 = new BinomialTreeNode(420);
+        assertEquals(treeNode3, BinomialTreeNode.merge(treeNode4, treeNode3), "You returned the wrong node!");
+        assertEquals(1, treeNode3.rank());
+        assertEquals(12, treeNode3.min());
+        assertEquals(treeNode3, BinomialTreeNode.merge(treeNode1, treeNode3), "You returned the wrong node!");
+        assertEquals(2, treeNode3.rank());
+
```

Comment:
Ther could be a min() test here.

Reply:
Yeah, I forgot about it, but have fixed now, thanks for pointing it out